source/features/git-checkout-pr.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import ClippyIcon from 'octicon/clippy.svg';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {getCurrentBranch} from '../github-helpers';
+
+const connectionType: Record<string, string> = {
+	HTTPS: `${location.origin}/`,
+	SSH: `git@${location.hostname}:`
+};
+
+function checkoutOption(option: string): JSX.Element {
+	const [, user, repository] = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.pathname.split('/', 3);
+	const isLocalPR = option === 'local';
+	return (
+		<>
+			{isLocalPR || <p className="text-gray text-small my-1">{option}</p>}
+			<div className="copyable-terminal">
+				<div className="copyable-terminal-button">
+					<clipboard-copy
+						className="btn btn-sm zeroclipboard-button"
+						role="button"
+						for={`rgh-checkout-pr-${option}`}
+						aria-label="Copy to clipboard"
+						data-copy-feedback="Copied!"
+					>
+						<ClippyIcon/>
+					</clipboard-copy>
+				</div>
+				<pre
+					id={`rgh-checkout-pr-${option}`}
+					className="copyable-terminal-content"
+				>
+					<span className="user-select-contain">
+						{isLocalPR || `git remote add ${user} ${connectionType[option]}${user}/${repository}.git\n`}
+						git fetch {isLocalPR ? 'origin' : user} {getCurrentBranch()}{'\n'}
```

Comment:
isn't `origin` for the owner and others `upstream`?

Reply:
It's the other way around: For me, https://github.com/FloEdelmann/refined-github is `origin` and https://github.com/sindresorhus/refined-github is `upstream`.

https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/github-glossary#upstream

Reply:
@FloEdelmann we are saying the same thing, you just said it better. 😀

Reply:
List every situation:

- owner fetches own repo: covered
- owner fetches third party repo: covered
- author fetches own PR: not covered
- anyone fetches PR: not covered

Any more cases?

Reply:
Not that I can think of.

Reply:
- author fetches own PR: not covered

Does not need to be covered IMHO

- anyone fetches third party PR: not covered

how do we determine who is anyone?

I would love to finish up this PR tonight

Reply:
It's kinda complicated and needs a little guessing, actually. Examples:
- as a collaborator of this repo, `origin` points to this repo
- you're also a collaborator of this repo, but I don't think that `origin` points to this repo yet since you still open

In short, if I see a PR opened from a branch here, I can pull `origin`, but if you see it, you will pull `upstream`, I'm guessing.

I think it varies based on the viewer and on the `head` repo:

<table>
<tr>
	<th>viewer ⬇️ 
	<th>head:<br>viewer’s own repo
	<th>head:<br>owner’s repo
	<th>head:<br>other repo
<tr>
	<th>owner
	<td colspan=2><code>origin</code>
	<td>$author
<tr>
	<th>collaborator
	<td><code>origin</code>
	<td><code>origin</code> or <s><code>upstream</code></s>
	<td>$author
<tr>
	<th>3rd party
	<td><code>origin</code>
	<td><s><code>origin</code></s> or <code>upstream</code>
	<td>$author
</table>

As you can see, there are 2 "or"s, but I crossed the less likely one.

Reply:
doing it now

Reply:
Check it out :)

Reply:
`getCollaborators` always includes you

Also the logic is different, `isLocal` is irrelevant.

Reply:
I think this is the code to use:


```js
// Logic found on https://github.com/sindresorhus/refined-github/pull/3596#discussion_r511546864
function getRemoteName() {
	const author = getPRRepositoryInfo().owner;
	if (author === getUsername()) {
		return; // `origin`, don't add remote
	}

	if (author !== getRepositoryInfo().owner) {
		return author;
	}

	if (select('[aria-label="Edit Pull Request title"]')) {
		return; // It's a collaborator, it's likely to be `origin`
	}

	return 'upstream';
}
```

Reply:
🎉 Thanks I have something similar. But your is much better!

Reply:
Again, isLocalPR is completely irrelevant and should be removed. Follow the table. Test it. I gave you all the information that’s needed. We’re pair programming again.

Reply:
My function is all that’s needed. If the function returns a string, prepend `git remote add`. If not, don’t add it.